proxy/logging/LogFilter.h

```diff
@@ -401,37 +401,54 @@ wipeField(char **field, char *pattern, const char *uppercase_field)
     }
 
     const char *p1 = strstr(lookup_query_param, pattern);
-    int field_pos  = p1 - lookup_query_param;
-    p1             = query_param + field_pos;
-
-    if (p1) {
-      char tmp_text[strlen(buf_dest) + 10];
-      char *temp_text = tmp_text;
-      memcpy(temp_text, buf_dest, (p1 - buf_dest));
-      temp_text += (p1 - buf_dest);
-      const char *p2 = strstr(p1, "=");
-      if (p2) {
-        p2++;
-        memcpy(temp_text, p1, (p2 - p1));
-        temp_text += (p2 - p1);
-        const char *p3 = strstr(p2, "&");
-        if (p3) {
-          for (int i = 0; i < (p3 - p2); i++) {
-            temp_text[i] = 'X';
+    if (!p1) {
+      return;
+    } else {
+      while(p1) {
```

Comment:
It seems like it'd probably be cleaner and more readable, if we move the code in the while loop into a small utility function (as it has grown too long now :) )

Reply:
Fixed it.